src/shape.cpp

```diff
@@ -361,31 +361,25 @@ std::size_t shape::index(std::size_t i) const
     }
 }
 
-std::vector<std::size_t> shape::multi(std::size_t i) const
+std::vector<std::size_t> shape::multi(std::size_t idx) const
 {
-    assert(this->standard());
-
+    if(idx >= elements())
```

Comment:
Add a throw for if the shape is dynamic.

Reply:
Call to `elements()` would throw in that case. I can update test.